optuna/integration/sklearn.py

```diff
@@ -398,7 +398,14 @@ class OptunaSearchCV(BaseEstimator):
             estimator supports ``partial_fit``.
 
         n_jobs:
-            Number of parallel jobs. :obj:`-1` means using all processors.
+            Number of :obj:`threading` based parallel jobs. :obj:`-1` means
+            using the number is set to CPU count.
+
+                .. note::
+                    ``n_jobs`` allows parallelization using :obj:`threading` and may suffer from
+                    `Python's GIL <https://wiki.python.org/moin/GlobalInterpreterLock>`_.
+                    It is recommended to use :ref:`process-based parallelization<distributed>`
+                    if ``func`` is CPU bound.
```

Comment:
Since `OptunaSearchCV` does not provide the multi-processing option, I couldn't find an easy way to apply process-based parallelization to it. How about just mentioning that multi-theading based parallelization is not suitable if ``func`` is CPU-bound?

Reply:
Thank you for your feedback. I supposed that `OptunaSearchCV` can be parallelized by using [distributed optimization](https://optuna.readthedocs.io/en/v2.6.0/tutorial/10_key_features/004_distributed.html#sphx-glr-tutorial-10-key-features-004-distributed-py) by following the note of `study.optimize` https://github.com/optuna/optuna/blob/d8a1a58b68e11ab9ea01631693bdbb0173204942/optuna/study.py#L351-L355

Reply:
Oh, I overlooked the link to [distributed optimization tutorial](https://optuna.readthedocs.io/en/v2.6.0/tutorial/10_key_features/004_distributed.html#sphx-glr-tutorial-10-key-features-004-distributed-py). I should have checked the rendered doc. Thank you.